src/test/java/core/basesyntax/strategy/PurchaseOperationTest.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseOperationTest {
+    private FruitOperationHandler purchaseOperation;
+    private Map<String, Integer> inventory;
+
+    @BeforeEach
+    void setUp() {
+        inventory = new HashMap<>();
+        purchaseOperation = new PurchaseOperation();
+    }
+
+    @Test
+    void addQuantityToFruit_executeOperation_Ok() {
+        inventory.put("apple", 5);
+        FruitTransaction transaction = new FruitTransaction(Operation.PURCHASE, "apple", 3);
+        purchaseOperation.executeOperation(transaction, inventory);
+
+        Assertions.assertEquals(2, inventory.get("apple"));
+    }
+
+    @Test
+    void subtractionOfProductThatAreLessThan0_executeOperation_Ok() {
+        FruitTransaction transaction = new FruitTransaction(Operation.PURCHASE, "apple", 10);
+        Assertions.assertThrows(RuntimeException.class,
+                () -> purchaseOperation.executeOperation(transaction, inventory));
+    }
```

Comment:
The assertThrows is not checking for any specific error message. It's a good practice to also check that the correct message is being thrown, not just the type of exception. You could do this by adding an additional parameter to the assertThrows method.